internal/p2p/peer_manager.go

```diff
@@ -0,0 +1,182 @@
+package p2p
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/libsv/go-p2p/wire"
+)
+
+var ErrPeerNetworkMismatch = fmt.Errorf("peer network mismatch")
+
+type PeerManager struct {
+	execWg        sync.WaitGroup
+	execCtx       context.Context
+	cancelExecCtx context.CancelFunc
+
+	l       *slog.Logger
+	network wire.BitcoinNet
+
+	mu    sync.RWMutex
+	peers []PeerI
+
+	restartUnhealthyPeers bool
+}
+
+func NewBetterPeerManager(logger *slog.Logger, network wire.BitcoinNet, options ...PeerManagerOptions) *PeerManager {
```

Comment:
you can use `NewPeerManager`

Reply:
Agree. Let's remove "better" everywhere.
We already know it's better ;)

Reply:
Done